### PR TITLE
Update to MultivariatePolynomials v0.5

### PR DESCRIPTION
**Pull request by blegat**
Most of the breaking changes were done in order to make it faster notably to resolve https://github.com/JuliaAlgebra/MultivariatePolynomials.jl/issues/194 (I should update the benchmarks in this issue by the way, they should be much better now).
The performance of multivariate gcd should be much better now.

### CODE REVIEW

**Comment by shashi** [comment]
CI is erroring with:

```
  `variable_union_type` not implemented for MultivariatePolynomials.AbstractVariable.
```

File: src/polyform.jl [hunk 1]
```diff
@@ -156,7 +155,7 @@ end
 function PolyForm(x,
         pvar2sym=get_pvar2sym(),
         sym2term=get_sym2term(),
-        vtype=DynamicPolynomials.PolyVar{true};
+        vtype=MP.AbstractVariable;
```
Comment (blegat) [inline src/polyform.jl:158]:
```suggestion
        vtype=DynamicPolynomials.Variable{ DynamicPolynomials.Commutative{DynamicPolynomials.CreationOrder},DynamicPolynomials.Graded{MP.LexOrder}};
```

**Comment by schillic** [comment]
If this looks good to you, could you please merge and tag a new release? This is holding back changes in other packages.

**Comment by shashi** [comment]
Thanks!

**Pull request merged**